hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/sender/http/OAuth2AuthProvider.java

```diff
@@ -0,0 +1,65 @@
+package pl.allegro.tech.hermes.consumers.consumer.sender.http;
+
+import pl.allegro.tech.hermes.api.OAuth2AuthenticationData;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.common.cache.zookeeper.NodeCache;
+import pl.allegro.tech.hermes.consumers.consumer.ConsumerMessageSender;
+import pl.allegro.tech.hermes.consumers.consumer.Message;
+
+import com.github.scribejava.core.builder.api.DefaultApi20;
+
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.oauth.OAuth20Service;
+import com.github.scribejava.core.model.OAuthConfig;
+
+public class OAuth2AuthProvider implements HttpAuthorizationProvider {
```

Comment:
i think we could use some metrics here - so we know how does the auth service behave, what are the response times etc, might come in handy :)